### ISSUE TITLE
Training on custom dataset

### ISSUE THREAD

**Issue by Shanki1** [pythonlessons/TensorFlow-2.x-YOLOv3#1]
Hi 
Great work sir.

while training on a class from google open dataset

have been facing this following issue 

epoch: 1 step: 3432/3434, lr:0.000100, giou_loss:   1.99, conf_loss:   5.01, prob_loss:  10.97, total_loss:  17.97
epoch: 1 step: 3433/3434, lr:0.000100, giou_loss:   2.27, conf_loss:   6.15, prob_loss:  11.99, total_loss:  20.41
epoch: 1 step:    0/3434, lr:nan, giou_loss:   2.14, conf_loss:   6.32, prob_loss:  12.02, total_loss:  20.48
epoch: 1 step:    1/3434, lr:nan, giou_loss:   2.78, conf_loss:   5.77, prob_loss:  13.06, total_loss:  21.61

giou_val_loss:    nan, conf_val_loss:    nan, prob_val_loss:    nan, total_val_loss:    nan

**Comment by pythonlessons** [comment]
giou_val_loss is for testing data, this means you don't have it or you forgot to change this in settings

**Comment by Shanki1** [comment]
Hi,
 
( by settings you mean in the yolov3/configs.py )

These are the following settings for test data

TEST_ANNOT_PATH             = "./OIDv4_ToolKit/OID/Dataset/Dataset_test.txt"
TEST_BATCH_SIZE             = 4
TEST_INPUT_SIZE             = 416
TEST_DATA_AUG               = False
TEST_DECTECTED_IMAGE_PATH   = "./data/detection/"
TEST_SCORE_THRESHOLD        = 0.3
TEST_IOU_THRESHOLD          = 0.45

My Dataset_test.txt file looks like this 
PS: PATH is the file path for jpg's in the test directory 

{PATH} 1,539,722,1022,0
{PATH} 0,0,1024,768,0
{PATH} 150,30,863,660,1
{PATH} 104,442,593,836,1

**Comment by pythonlessons** [comment]
Yes, yolov3/configs.py.
But everything seems fine...
Did you received same NaN while training mnist ?

**Comment by Shanki1** [comment]
no i did not received NaN while training mnist.
now i trained the model for two classes  which works fine 

i have no clue why is it not even creating a model when trained on all classes 

running train.py again ( in process )

**Comment by pythonlessons** [comment]
Check your classes, if your class has space, it may be bad, for example, "Fire hydrant" should be "Fire_hydrant" also the same for dataset folder names.
I am also not sure why you may receive this problem...

**Comment by Shanki1** [comment]
yes i already checked the spaces as it is properly done 
but somewhere i suspect this class 'Kitchen_&_dining_room_table'

OIDv4_ToolKit/OID/Dataset/test/Kitchen_&_dining_room_table/4e473fdc461aac33.jpg  0,1,1024,659,13

again it didn't save the model 
it executed the same way as before.

**Comment by pythonlessons** [comment]
Can you try changing "Kitchen_&_dining_room_table" to something simpler for testing? if this is the cause.

What classes did you download from google open dataset? would like to try getting the same issue

**Comment by Shanki1** [comment]
I tried changing the name and it still did not work.
 
It works fine till the end but does not save the model.

These are the following classes that i am using (29 classes) :

Swimming_pool
Kitchen_dining_room_table ( i renamed it but original is Kitchen_&_dining_room_table)
Fireplace
Jacuzzi
Mirror
Couch
Pillow
Refrigerator
Dishwasher
Microwave_oven
Porch
Washing_machine
Ceiling_fan
Sink
Gas_stove
Tree_house
Coffeemaker
Toilet
Sofa_bed
Oven
Towel
Billiard_table
Countertop
Bathtub
Shower
Television
Stairs
Fountain
Bed

**Comment by Shanki1** [comment]
Hello Sir,
Are you trying with the above classes ?

**Comment by pythonlessons** [comment]
> Hello Sir,
> Are you trying with the above classes ?

Hi, 
I was checking for some dataset to download from https://storage.googleapis.com/openimages/web/index.html but couldn't find them. Maybe you used different sources to get them?

**Comment by pythonlessons** [comment]
Did you have a test dataset together with the train dataset?
Try to decrease learning rate

**Comment by Shanki1** [comment]
Hi i tried with the same classes as you mentioned in your new video it worked fine but those classes are from OIDv6 and i have taken the classes from OIDv4 
i have no idea if that should affect but i will soon try taking similar classes from OIDv6 and execute.

**Comment by pythonlessons** [comment]
> Hi i tried with the same classes as you mentioned in your new video it worked fine but those classes are from OIDv6 and i have taken the classes from OIDv4
> i have no idea if that should affect but i will soon try taking similar classes from OIDv6 and execute.

Hi, nice to know that it worked, but in my understanding, you can't pick images from OIDv4 if there is OIDv6, toolkit downloads newest images. If I am wrong, fix me.

**Comment by pythonlessons** [comment]
If it worked fine, I close this
